mmdet/datasets/pipelines/auto_augment.py

```diff
@@ -69,4 +72,193 @@ def __call__(self, results):
+            fill_val = tuple([float(fill_val)])
+        elif isinstance(fill_val, tuple):
+            assert len(fill_val) == 3, \
+                'fill_val as tuple must have 3 elements.'
+            fill_val = tuple([float(val) for val in fill_val])
+        else:
+            raise ValueError(
+                'fill_val must be a float scale or a tuple with 3 elements.')
+        assert np.all([0 <= val <= 255 for val in fill_val]), \
+            'all elements of fill_val should between range [0,255].'
+        assert axis in ('x', 'y'), \
+            'Translate should be alone with x-axis or y-axis.'
+        assert isinstance(max_translate_offset, (int, float)), \
+            'The max_translate_offset must be type int or float.'
+        # the offset for translation
+        self.offset = int(
+            (level / Translate._MAX_LEVEL) * max_translate_offset)
+        self.level = level
+        self.prob = prob
+        self.fill_val = fill_val
+        self.axis = axis
+        self.max_translate_offset = max_translate_offset
+
+    @staticmethod
+    def warpAffine(data,
+                   trans_matrix,
+                   out_size,
+                   fill_val,
+                   flags=cv2.INTER_NEAREST,
+                   borderMode=cv2.BORDER_CONSTANT):
+        return cv2.warpAffine(
+            data,
+            trans_matrix,
+            dsize=out_size,  # dsize takes input size as order (w,h).
+            flags=flags,
+            borderMode=borderMode,
+            borderValue=fill_val)
+
+    def __call__(self, results, min_size=0.0, neg_offset_prob=0.5):
+        if np.random.rand() > self.prob:
+            return results
+
+        offset = -self.offset if np.random.rand(
+        ) < neg_offset_prob else self.offset
+        # the transformation matrix of cv2
+        if self.axis == 'x':
+            trans_matrix = np.float32([[1, 0, offset], [0, 1, 0]])
+        else:
+            trans_matrix = np.float32([[1, 0, 0], [0, 1, offset]])
+
+        self._translate_img(results, trans_matrix, fill_val=self.fill_val)
+        self._translate_bboxes(results, offset)
+        # fill_val set to 0 for background of mask.
+        self._translate_masks(results, trans_matrix, fill_val=0)
+        # fill_val set to 255 for the ignored value of segmentation map.
+        self._translate_seg(results, trans_matrix, fill_val=255)
+        self._filter_invalid(results, min_size=min_size)
+        return results
+
+    def _translate_img(self, results, trans_matrix, fill_val):
+        for key in results.get('img_fields', ['img']):
+            results[key] = self.warpAffine(results[key], trans_matrix,
+                                           results[key].shape[:2][::-1],
+                                           fill_val)
+
+    def _translate_bboxes(self, results, offset):
+        h, w, c = results['img_shape']
+        for key in results.get('box_fields', []):
+            min_x, min_y, max_x, max_y = results[key]
```

Comment:
`box_fields` usually contains an array of bboxes. So I guess `min_x, min_y, max_x, max_y = results[key]` will report an error here.